pkg/asset/manifests/clusterapi/cluster.go

```diff
@@ -0,0 +1,230 @@
+package clusterapi
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/asset/ignition/machine"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/manifests"
+	"github.com/openshift/installer/pkg/asset/manifests/aws"
+	"github.com/openshift/installer/pkg/asset/manifests/azure"
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	"github.com/openshift/installer/pkg/asset/openshiftinstall"
+	"github.com/openshift/installer/pkg/asset/rhcos"
+	"github.com/openshift/installer/pkg/clusterapi"
+)
+
+const (
+	manifestDir = "cluster-api-cluster-manifests"
+)
+
+var _ asset.WritableAsset = (*Cluster)(nil)
+
+// Cluster generates manifests for target cluster
+// creation using CAPI.
+type Cluster struct {
+	FileList  []*asset.File
+	Manifests capiutils.Manifests `json:"-"`
+}
+
+// Name returns a human friendly name for the operator.
+func (c *Cluster) Name() string {
+	return "Cluster API Manifests"
+}
+
+// Dependencies returns all of the dependencies directly needed by the
+// ClusterAPI asset.
+func (c *Cluster) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+		&installconfig.ClusterID{},
+		&openshiftinstall.Config{},
+		&manifests.FeatureGate{},
+		&bootstrap.Bootstrap{},
+		&machine.Master{},
+		new(rhcos.Image),
+	}
+}
+
+// Generate generates the respective operator config.yml files.
+func (c *Cluster) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	clusterID := &installconfig.ClusterID{}
+	openshiftInstall := &openshiftinstall.Config{}
+	featureGate := &manifests.FeatureGate{}
+	bootstrapIgnAsset := &bootstrap.Bootstrap{}
+	masterIgnAsset := &machine.Master{}
+	rhcosImage := new(rhcos.Image)
+	dependencies.Get(installConfig, clusterID, openshiftInstall, bootstrapIgnAsset, masterIgnAsset, featureGate, rhcosImage)
+
+	// If the feature gate is not enabled, do not generate any manifests.
+	if !capiutils.IsEnabled(installConfig) {
+		return nil
+	}
```

Comment:
needs #7413 

If you don't want to be blocked by #7413 (it just needs tags :sob: ) I will leave another comment for how to implement this

Reply:
```suggestion
	if !installConfig.EnabledFeatureGates().Enabled(v1.FeatureGateClusterAPIInstall) {
	    return nil
	}
```